packages/server/metricsHandler.ts

```diff
@@ -0,0 +1,186 @@
+
+const nodeActiveHandles = new Gauge({
+  name: 'node_active_handles',
+  help: 'Number of active handles',
+  registers: [register]
+})
+
+// WebSocket metrics
+export const wsConnections = new Gauge({
+  name: 'websocket_connections_total',
+  help: 'Total number of active WebSocket connections',
+  labelNames: ['port'],
+  registers: [register]
+})
+
+export const wsMessages = new Counter({
+  name: 'websocket_messages_total',
+  help: 'Total number of WebSocket messages',
+  labelNames: ['type', 'port'],
+  registers: [register]
+})
+
+export const wsMessageSize = new Histogram({
+  name: 'websocket_message_size_bytes',
+  help: 'Size of WebSocket messages in bytes',
+  labelNames: ['type', 'port'],
+  buckets: [100, 500, 1000, 5000, 10000, 50000, 100000],
+  registers: [register]
+})
+
+export const wsConnectionDuration = new Histogram({
+  name: 'websocket_connection_duration_seconds',
+  help: 'Duration of WebSocket connections in seconds',
+  labelNames: ['port'],
+  buckets: [60, 300, 600, 1800, 3600, 7200, 14400, 28800, 57600, 86400],
+  registers: [register]
+})
+
+export const wsErrors = new Counter({
+  name: 'websocket_errors_total',
+  help: 'Total number of WebSocket errors',
+  labelNames: ['type', 'port'],
+  registers: [register]
+})
+
+// Cache for event loop lag measurement
+let lastEventLoopLag = 0
+let lastEventLoopCheck = 0
+const EVENT_LOOP_CHECK_INTERVAL = 1000
+
+// GraphQL Yoga Prometheus plugin configuration
+export const graphqlPrometheusPlugin = usePrometheus({
+  registry: register
+})
+
+const updateWebSocketMetrics = () => {
+  wsConnections.set({port: PORT.toString()}, activeClients.size)
+}
+
+const updateNodeMetrics = () => {
+  const memoryUsage = process.memoryUsage()
+  const {heapTotal, heapUsed, external, rss, arrayBuffers} = memoryUsage
+
+  nodeMemoryUsage.set({type: 'heapTotal'}, heapTotal)
+  nodeMemoryUsage.set({type: 'heapUsed'}, heapUsed)
+  nodeMemoryUsage.set({type: 'external'}, external)
+  nodeMemoryUsage.set({type: 'rss'}, rss)
+  nodeMemoryUsage.set({type: 'arrayBuffers'}, arrayBuffers)
+
+  nodeHeapSize.set({type: 'total'}, heapTotal)
+  nodeHeapSize.set({type: 'used'}, heapUsed)
+
+  const cpuUsage = process.cpuUsage()
+  const userCpu = cpuUsage.user / 1_000_000
+  const systemCpu = cpuUsage.system / 1_000_000
+  nodeCpuUsage.set({type: 'user'}, userCpu)
+  nodeCpuUsage.set({type: 'system'}, systemCpu)
+
+  const now = Date.now()
+  if (now - lastEventLoopCheck >= EVENT_LOOP_CHECK_INTERVAL) {
+    const start = process.hrtime.bigint()
+    setTimeout(() => {
+      const end = process.hrtime.bigint()
+      lastEventLoopLag = Number(end - start) / 1_000_000_000
+      lastEventLoopCheck = now
+    }, 0)
+  }
+  nodeEventLoopLag.set(lastEventLoopLag)
+
+  const activeResources = process.getActiveResourcesInfo()
+  const gcCount = activeResources.reduce(
+    (count, resource) => (resource.includes('gc') ? count + 1 : count),
+    0
+  )
+  nodeGcDuration.observe({type: 'total'}, gcCount)
```

Comment:
-1 now we're not observing this over time anymore. Shouldn't this run on a timer?